linera-views/src/reentrant_collection_view.rs

```diff
@@ -506,17 +514,33 @@ where
                 btree_map::Entry::Occupied(entry) => {
                     let entry = entry.into_mut();
                     if let Update::Removed = entry {
-                        selected_short_keys.push((short_key, true));
+                        let key = self
+                            .context
+                            .base_tag_index(KeyTag::Subview as u8, &short_key);
+                        let context = self.context.clone_with_base_key(key);
+                        let view = W::new(context)?;
+                        let view = Arc::new(RwLock::new(view));
+                        *entry = Update::Set(view);
                     }
                 }
-                btree_map::Entry::Vacant(_entry) => {
-                    selected_short_keys.push((short_key, self.delete_storage_first));
+                btree_map::Entry::Vacant(entry) => {
+                    if self.delete_storage_first {
+                        let key = self
+                            .context
+                            .base_tag_index(KeyTag::Subview as u8, &short_key);
+                        let context = self.context.clone_with_base_key(key);
+                        let view = W::new(context)?;
+                        let view = Arc::new(RwLock::new(view));
+                        entry.insert(Update::Set(view));
+                    } else {
+                        selected_short_keys.push(short_key);
+                    }
                 }
             }
         }
         let mut handles = Vec::new();
         for short_key in &selected_short_keys {
-            let short_key = short_key.0.clone();
+            let short_key = short_key.clone();
```

Comment:
I can't see where we're using this clone. Move it to where it's used?

Reply:
Exactly, when the move happens, the clone is needed.

Reply:
Yes, I could remove the big clone operation.